platform-sdk/platform-apps/tests/StatsSigningTestingTool/src/main/java/com/swirlds/demo/stats/signing/StatsSigningTestingToolState.java

```diff
@@ -175,6 +192,36 @@ private void handleTransaction(final ConsensusTransaction trans) {
         maybeDelay();
     }
 
+    /**
+     * Checks if the transaction bytes are system ones.
+     *
+     * @param transaction the transaction to check
+     * @return true if the transaction bytes are system ones, false otherwise
+     */
+    private boolean areTransactionBytesSystemOnes(final Transaction transaction) {
+        final var transactionBytes = transaction.getApplicationTransaction();
+
+        if (transactionBytes.length() == 0) {
+            return false;
+        }
+
+        final var transactionPool = transactionPoolSupplier.get();
+
+        // Check the data length size is the expected size based on the transactionSize property.
+        // If it is not, then the transaction is a system one.
+        if (TransactionCodec.txIsSigned(transactionBytes)) {
+            // Data length can be read directly from the extracted signature
+            final var signature = TransactionCodec.extractSignature(transactionBytes);
+            return signature.getMessageLength() != transactionPool.getTransactionSize();
+        } else {
+            // Data length value can be read from the transaction bytes using an offset
+            final ByteBuffer wrapper =
+                    ByteBuffer.wrap(transactionBytes.toByteArray()).position(21);
+            final byte dataLength = wrapper.get();
+            return dataLength != transactionPool.getTransactionSize();
+        }
+    }
+
```

Comment:
this does not work if they happen to be the same size, so we need a definitive way of differentiating them

Reply:
Added additional logic that follows the transaction bytes structure from `TransactionCodec`. Now the algorithm checks if we have the correct size for the different segments on the correct positions and whether we have any bytes left after we load the `data segment` (which is the last one), using the `data length` value before it. I believe it will be very unlikely to have a system transaction that will cover all the additional checks I've added.

Another approach, which will be more harming for the performance, is trying to parse the transactions bytes to a `StateSignatureTransaction` pbj type for each incoming transaction.